api-put-bucket.go

```diff
@@ -477,3 +479,95 @@ func (c Client) EnableVersioning(ctx context.Context, bucketName string) error {
 func (c Client) DisableVersioning(ctx context.Context, bucketName string) error {
 	return c.setVersioning(ctx, bucketName, versionDisableConfig, versionDisableConfigLen, versionDisableConfigMD5Sum, versionDisableConfigSHA256)
 }
+
+// RemoveBucketReplication removes a replication config on an existing bucket.
+func (c Client) RemoveBucketReplication(ctx context.Context, bucketName string) error {
+	return c.SetBucketReplication(ctx, bucketName, replication.Config{}, "")
+}
+
+// SetBucketReplication sets a replication config on an existing bucket.
+func (c Client) SetBucketReplication(ctx context.Context, bucketName string, cfg replication.Config, replicationArn string) error {
+	// Input validation.
+	if err := s3utils.CheckValidBucketName(bucketName); err != nil {
+		return err
+	}
+
+	// If replication is empty then delete it.
+	if cfg.Empty() {
+		return c.removeBucketReplication(ctx, bucketName)
+	}
+	if replicationArn == "" {
+		return fmt.Errorf("Replication ARN not set")
+	}
+	// Save the updated replication.
+	return c.putBucketReplication(ctx, bucketName, cfg, replicationArn)
+}
+
+// Saves a new bucket replication.
+func (c Client) putBucketReplication(ctx context.Context, bucketName string, cfg replication.Config, replicationArn string) error {
+	// Input validation.
+	if err := s3utils.CheckValidBucketName(bucketName); err != nil {
+		return err
+	}
+
+	// Get resources properly escaped and lined up before
+	// using them in http request.
+	urlValues := make(url.Values)
+	urlValues.Set("replication", "")
+	headers := make(http.Header)
+	headers.Add("x-minio-replication-arn", replicationArn)
```

Comment:
we should make this a const elsewhere in code to indicate why is this needed? since this is specific for MinIO?

Reply:
No longer needed as this ARN will be part of the configuration